geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -2109,24 +2109,26 @@ int entryCount(Set<Integer> buckets, boolean estimate) {
   @Override
   public int getRegionSize() {
     synchronized (getSizeGuard()) {
-      int result = getRegionMap().size();
-      // if this is a client with no tombstones then we subtract the number
-      // of entries being affected by register-interest refresh
-      if (imageState.isClient() && !getConcurrencyChecksEnabled()) {
-        int destroyedEntriesCount = imageState.getDestroyedEntriesCount();
-        if (result < destroyedEntriesCount) {
-          logger.error("Incorrect region size: mapSize={}, destroyedEntriesCount={}.", result,
-              destroyedEntriesCount);
+      synchronized (tombstoneCount) {
```

Comment:
I think synchronizing on an atomic is an "odd" thing to do and is worth you adding a comment here to explain the race it prevents. I think it would even be worth having the comment refer to your new regionSizeShouldAlwaysBePositive test